src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java

```diff
@@ -176,9 +217,470 @@ private Object createObject(String className) {
             declaredConstructor.setAccessible(true);
             instance = declaredConstructor.newInstance();
         }
-        catch (final ReflectiveOperationException | NoClassDefFoundError exception) {
-            LOG.debug(IGNORING_EXCEPTION_MESSAGE, exception);
+        catch (final ReflectiveOperationException | NoClassDefFoundError ignored) {
+            // keep looking, ignoring exception
         }
         return instance;
     }
+
+    /**
+     * Fill short-to-full module names map.
+     */
+    private static void fillShortToFullModuleNamesMap() {
+        fillChecksFromAnnotationPackage();
+        fillChecksFromBlocksPackage();
+        fillChecksFromCodingPackage();
+        fillChecksFromDesingPackage();
+        fillChecksFromHeaderPackage();
+        fillChecksFromImportsPackage();
+        fillChecksFromIndentationPackage();
+        fillChecksFromJavadocPackage();
+        fillChecksFromMetricsPackage();
+        fillChecksFromModifierPackage();
+        fillChecksFromNamingPackage();
+        fillChecksFromRegexpPackage();
+        fillChecksFromSizesPackage();
+        fillChecksFromWhitespacePackage();
+        fillModulesFromChecksPackage();
+        fillModulesFromFilefiltersPackage();
+        fillModulesFromFiltersPackage();
+        fillModulesFromCheckstylePackage();
+    }
+
+    /**
+     * Fill short-to-full module names map with Checks from annotation package.
+     */
+    private static void fillChecksFromAnnotationPackage() {
+        NAME_TO_FULL_MODULE_NAME.put("AnnotationLocationCheck",
+                "com.puppycrawl.tools.checkstyle.checks.annotation.AnnotationLocationCheck");
```

Comment:
To me, it seems that if we use this field some of the time, we should use it all of the time. It looks odd to see it switch back and forth.
It is sorta similar to `MultipleStringLiteralsCheck` except the string is a sub-string of another instead of an exact duplicate.